### DISCUSSION TITLE
Source code authorship attribution comments in PSyclone

### DISCUSSION THREAD

**Discussion by nmnobre** [stfc/PSyclone#1995] [Polls]
Hi everyone!

**Comment by sergisiso** [comment]
The cons are that they are cumbersome to maintain, they are not always accurate (e.g forgotten updates, copy/paste between files, ... ) and are not legally required for someone to be recognized as a copyright holder (the actual file history is).
The pros are that BSD-3 requires attribution, so it makes it easier for people who want to take/fork a single file or part of the project to keep the attribution without doing extra work.

An alternative to satisfy pros without cons would be to add a AUTHORS global file and on each file say:

```
# Part of the PSyclone Project, released under the BSD 3-Clause License.
# See LICENSE or go to https://raw.githubusercontent.com/stfc/PSyclone/master/LICENSE for license
# information. See AUTHORS or go to https://raw.githubusercontent.com/stfc/PSyclone/master/AUTHORS
# for authorship information.
```

This is what some projects like LLVM do.

**Reply by nmnobre** [reply]
STFC is the sole copyright holder according to the license text, individuals are only "contributors" but hold no copyright.
I would much prefer a single list of authors in the root of the repo like you suggest.

**Reply by sergisiso** [reply]
This is probably not the case whatever the license text says (https://opensource.stackexchange.com/questions/11194/can-a-cla-be-agreed-to-implicitly-like-a-license). In general the copyright is owned by the person/institution who create those lines unless a CLA is signed. However, this doesn't matter for BSD-3 open-source license because the rights are the same either way.

**Reply by sergisiso** [reply]
I discussed this with @arporter and we settled on a shorter, SPDX-based header with no authors:
```
# -----------------------------------------------------------------------------
# SPDX-FileCopyrightText: Copyright (c) 2017-2026 Science and Technology
#                         Facilities Council
# SPDX-License-Identifier: BSD-3-Clause
# See the full LICENSE file in the project root for details.
# -----------------------------------------------------------------------------
```

**Comment by sergisiso** [comment]
@arporter We need to discuss how we replace the text around inexistent/private links:

```
! For further details please refer to Met Office Science Repository Service:
! https://code.metoffice.gov.uk/trac/lfric/browser/LFRic/trunk/LICENCE.original
```
and
```
#                         However, it has been created with the help of the
#                         GungHo Consortium, whose members are identified at
#                         https://puma.nerc.ac.uk/trac/GungHo/wiki
```

The current suggestion is replacing the first to LFRic_core/apps and deleting the second from  ", whose"
